contracts/interchain-token/tests/test.rs

```diff
@@ -4,55 +4,515 @@ extern crate std;
+    assert_invoke_auth_ok!(minter1, token.try_mint(&minter1, &user, &amount));
+
+    assert_eq!(
+        env.auths(),
+        std::vec![(
+            minter1.clone(),
+            AuthorizedInvocation {
+                function: AuthorizedFunction::Contract((
+                    token.address.clone(),
+                    symbol_short!("mint"),
+                    (minter1.clone(), &user, amount).into_val(&env),
+                )),
+                sub_invocations: std::vec![]
+            }
+        )]
+    );
+    assert_eq!(token.balance(&user), amount);
+
+    assert_invoke_auth_ok!(owner, token.try_remove_minter(&minter1));
 
     assert_last_emitted_event(
         &env,
-        &client.address,
+        &token.address,
+        (Symbol::new(&env, "minter_removed"), minter1.clone()),
+        (),
+    );
+
+    assert_invoke_auth_err!(minter1, token.try_mint(&minter1, &user, &amount));
+}
+
+#[test]
+fn test_burn() {
+    let env = Env::default();
+
+    let owner = Address::generate(&env);
+    let minter = Address::generate(&env);
+    let user1 = Address::generate(&env);
+    let user2 = Address::generate(&env);
+    let token = create_token(&env, &owner, &minter);
+    let amount = 1000;
+
+    assert_invoke_auth_ok!(minter, token.try_mint(&minter, &user1, &amount));
+    assert_eq!(token.balance(&user1), amount);
+
+    let expiration_ledger = 200;
+    let amount2 = 500;
+
+    assert_invoke_auth_ok!(
+        user1,
+        token.try_approve(&user1, &user2, &amount2, &expiration_ledger)
+    );
+    assert_eq!(token.allowance(&user1, &user2), amount2);
+
+    assert_invoke_auth_ok!(user2, token.try_burn_from(&user2, &user1, &amount2));
+    assert_eq!(
+        env.auths(),
+        std::vec![(
+            user2.clone(),
+            AuthorizedInvocation {
+                function: AuthorizedFunction::Contract((
+                    token.address.clone(),
+                    symbol_short!("burn_from"),
+                    (&user2, &user1, amount2).into_val(&env),
+                )),
+                sub_invocations: std::vec![]
+            }
+        )]
+    );
+
+    assert_eq!(token.allowance(&user1, &user2), 0);
+    assert_eq!(token.balance(&user1), amount2);
+    assert_eq!(token.balance(&user2), 0);
+
+    assert_invoke_auth_ok!(user1, token.try_burn(&user1, &amount2));
+    assert_eq!(
+        env.auths(),
+        std::vec![(
+            user1.clone(),
+            AuthorizedInvocation {
+                function: AuthorizedFunction::Contract((
+                    token.address.clone(),
+                    symbol_short!("burn"),
+                    (&user1, amount2).into_val(&env),
+                )),
+                sub_invocations: std::vec![]
+            }
+        )]
+    );
+
+    assert_eq!(token.balance(&user1), 0);
+    assert_eq!(token.balance(&user2), 0);
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Context, InvalidAction)")]
+fn decimal_is_over_max() {
```

Comment:
```suggestion
fn register_token_with_invalid_decimals_fails() {
```
the function being tested should be clear from the test name

Reply:
updated